onig_sys doesn't build with musl

[muslrust](https://github.com/clux/muslrust#readme) is a Docker image for building Rust projects using musl libc. The nice thing about it is that it can produce statically linked binaries that don't have any dependencies (not even to libc).

It looks like rust-onig currently fails to link in muslrust:

```
$ docker run -v $PWD:/volume -w /volume -t clux/muslrust cargo build --release
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading onig v1.1.0
 Downloading libc v0.2.16
 Downloading onig_sys v61.1.0
 Downloading bitflags v0.7.0
 Downloading lazy_static v0.2.1
 Downloading pkg-config v0.3.8
 Downloading cmake v0.1.17
 Downloading gcc v0.3.35
   Compiling bitflags v0.7.0
   Compiling lazy_static v0.2.1
   Compiling pkg-config v0.3.8
   Compiling gcc v0.3.35
   Compiling libc v0.2.16
   Compiling cmake v0.1.17
   Compiling onig_sys v61.1.0
error: failed to run custom build command for `onig_sys v61.1.0`
process didn't exit successfully: `/volume/target/release/build/onig_sys-1a89ba0e5167b8c5/build-script-build` (exit code: 101)
--- stdout
running: "cmake" "/root/.cargo/registry/src/github.com-1ecc6299db9ec823/onig_sys-61.1.0/oniguruma" "-DCMAKE_MACOSX_RPATH=NO" "-DCMAKE_INSTALL_PREFIX=/volume/target/x86_64-unknown-linux-musl/release/build/onig_sys-1a89ba0e5167b8c5/out" "-DCMAKE_C_FLAGS= -O0 -ffunction-sections -fdata-sections -m64 -fPIC -static" "-DCMAKE_C_COMPILER=/usr/bin/musl-gcc" "-DCMAKE_CXX_FLAGS= -O0 -ffunction-sections -fdata-sections -m64 -fPIC -static" "-DCMAKE_CXX_COMPILER=musl-g++" "-DCMAKE_BUILD_TYPE=Release"
-- The C compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/musl-gcc
-- Check for working C compiler: /usr/bin/musl-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for alloca
-- Looking for alloca - not found
-- Looking for include file alloca.h
-- Looking for include file alloca.h - found
-- Looking for include file stdarg.h
-- Looking for include file stdarg.h - found
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file sys/times.h
-- Looking for include file sys/times.h - found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of short
-- Check size of short - done
-- Looking for 4 include files stdlib.h, ..., float.h
-- Looking for 4 include files stdlib.h, ..., float.h - found
-- Configuring done
-- Generating done
-- Build files have been written to: /volume/target/x86_64-unknown-linux-musl/release/build/onig_sys-1a89ba0e5167b8c5/out/build
running: "cmake" "--build" "." "--target" "install" "--config" "Release" "--" "-j4"
Scanning dependencies of target onig
[  1%] Building C object CMakeFiles/onig.dir/src/regerror.c.o
[  3%] Building C object CMakeFiles/onig.dir/src/regparse.c.o
[  5%] Building C object CMakeFiles/onig.dir/src/regext.c.o
[  7%] Building C object CMakeFiles/onig.dir/src/regcomp.c.o
[  9%] Building C object CMakeFiles/onig.dir/src/regexec.c.o
[ 11%] Building C object CMakeFiles/onig.dir/src/reggnu.c.o
[ 13%] Building C object CMakeFiles/onig.dir/src/regenc.c.o
[ 15%] Building C object CMakeFiles/onig.dir/src/regsyntax.c.o
[ 17%] Building C object CMakeFiles/onig.dir/src/regtrav.c.o
[ 19%] Building C object CMakeFiles/onig.dir/src/regversion.c.o
[ 21%] Building C object CMakeFiles/onig.dir/src/st.c.o
[ 23%] Building C object CMakeFiles/onig.dir/src/regposix.c.o
[ 25%] Building C object CMakeFiles/onig.dir/src/regposerr.c.o
[ 27%] Building C object CMakeFiles/onig.dir/src/onig_init.c.o
[ 29%] Building C object CMakeFiles/onig.dir/src/unicode.c.o
[ 31%] Building C object CMakeFiles/onig.dir/src/ascii.c.o
[ 33%] Building C object CMakeFiles/onig.dir/src/utf8.c.o
[ 35%] Building C object CMakeFiles/onig.dir/src/utf16_be.c.o
[ 37%] Building C object CMakeFiles/onig.dir/src/utf16_le.c.o
[ 39%] Building C object CMakeFiles/onig.dir/src/utf32_be.c.o
[ 41%] Building C object CMakeFiles/onig.dir/src/utf32_le.c.o
[ 43%] Building C object CMakeFiles/onig.dir/src/euc_jp.c.o
[ 45%] Building C object CMakeFiles/onig.dir/src/sjis.c.o
[ 47%] Building C object CMakeFiles/onig.dir/src/iso8859_1.c.o
[ 49%] Building C object CMakeFiles/onig.dir/src/iso8859_2.c.o
[ 50%] Building C object CMakeFiles/onig.dir/src/iso8859_3.c.o
[ 52%] Building C object CMakeFiles/onig.dir/src/iso8859_4.c.o
[ 54%] Building C object CMakeFiles/onig.dir/src/iso8859_5.c.o
[ 56%] Building C object CMakeFiles/onig.dir/src/iso8859_6.c.o
[ 58%] Building C object CMakeFiles/onig.dir/src/iso8859_7.c.o
[ 60%] Building C object CMakeFiles/onig.dir/src/iso8859_8.c.o
[ 62%] Building C object CMakeFiles/onig.dir/src/iso8859_9.c.o
[ 64%] Building C object CMakeFiles/onig.dir/src/iso8859_10.c.o
[ 66%] Building C object CMakeFiles/onig.dir/src/iso8859_11.c.o
[ 68%] Building C object CMakeFiles/onig.dir/src/iso8859_13.c.o
[ 70%] Building C object CMakeFiles/onig.dir/src/iso8859_15.c.o
[ 72%] Building C object CMakeFiles/onig.dir/src/iso8859_16.c.o
[ 74%] Building C object CMakeFiles/onig.dir/src/iso8859_14.c.o
[ 76%] Building C object CMakeFiles/onig.dir/src/euc_tw.c.o
[ 78%] Building C object CMakeFiles/onig.dir/src/euc_kr.c.o
[ 80%] Building C object CMakeFiles/onig.dir/src/big5.c.o
[ 82%] Building C object CMakeFiles/onig.dir/src/gb18030.c.o
[ 84%] Building C object CMakeFiles/onig.dir/src/koi8_r.c.o
[ 86%] Building C object CMakeFiles/onig.dir/src/cp1251.c.o
[ 88%] Building C object CMakeFiles/onig.dir/src/euc_jp_prop.c.o
[ 90%] Building C object CMakeFiles/onig.dir/src/sjis_prop.c.o
[ 92%] Building C object CMakeFiles/onig.dir/src/unicode_unfold_key.c.o
[ 94%] Building C object CMakeFiles/onig.dir/src/unicode_fold1_key.c.o
[ 96%] Building C object CMakeFiles/onig.dir/src/unicode_fold2_key.c.o
[ 98%] Building C object CMakeFiles/onig.dir/src/unicode_fold3_key.c.o
[100%] Linking C shared library libonig.so
CMakeFiles/onig.dir/build.make:1368: recipe for target 'libonig.so' failed
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/onig.dir/all' failed
Makefile:149: recipe for target 'all' failed

--- stderr
DIST_NAME: onig
DIST_VERSION: 6.1.1
DIST_LICENSE: BSD
DIST_AUTHOR: K.Kosako
DIST_MAINTAINER: K.Kosako
DIST_URL: https://github.com/kkos/oniguruma
DIST_DESC: Oniguruma is a regular expressions library.
DIST_DEPENDS: 
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_CXX_COMPILER
    CMAKE_CXX_FLAGS


/usr/bin/ld: /usr/lib/x86_64-linux-musl/libc.a(__stack_chk_fail.o): relocation R_X86_64_32 against `__stack_chk_guard' can not be used when making a shared object; recompile with -fPIC
/usr/lib/x86_64-linux-musl/libc.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
make[2]: *** [libonig.so] Error 1
make[1]: *** [CMakeFiles/onig.dir/all] Error 2
make: *** [all] Error 2
thread 'main' panicked at '
command did not execute successfully, got: exit code: 2

build script failed, must exit now', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.17/src/lib.rs:463
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

You should be able to reproduce it using [this test repo](https://github.com/robinst/rust-onig-test).

Would be very cool to make this work :).

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,9 +22,11 @@ libc = "0.2"
 bitflags = "0.7"
 lazy_static = "0.2"
 
-[dependencies.onig_sys]
+[target.'cfg(not(target_env = "musl"))'.dependencies.onig_sys]
 version = "61.1"
 path = "onig_sys"
 
-[target.'cfg(target_env = "musl")'.dependencies]
-onig_sys = { features = [ "static_onig"] }
+[target.'cfg(target_env = "musl")'.dependencies.onig_sys]
+features = [ "static_onig"]
+version = "61.1"
+path = "onig_sys"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,3 +25,6 @@ lazy_static = "0.2"
 [dependencies.onig_sys]
 version = "61.1"
 path = "onig_sys"
+
+[target.'cfg(target_env = "musl")'.dependencies]
+onig_sys = { features = [ "static_onig"] }

